celery/schedules.py

```diff
@@ -101,6 +101,11 @@ def to_local(self, dt):
             return timezone.to_local_fallback(dt)
         return dt
 
+    def __eq__(self, other):
+        if isinstance(other, BaseSchedule):
+            return other.nowfun == self.nowfun
```

Comment:
What I meant is that before this pull request, in celery beat, nowfun does not work at all.